src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1442,6 +1446,129 @@ def disable_features(cmd, client, resource_group_name, cluster_name, features, k
+            required_node_counts = 0
+            for item in api_response.items:
+                node_arch = item.metadata.labels.get("kubernetes.io/arch")
+                node_os = item.metadata.labels.get("kubernetes.io/os")
+                if node_arch == "amd64" and node_os == "linux":
+                    tr_logger.info("Node {} has capacity: CPU: {}, and Memory: {} ".format(required_node_counts + 1, item.status.capacity.get("cpu"), item.status.capacity.get("memory")))
+                    required_node_counts += 1
+            if required_node_counts == 0:
+                tr_logger.error("There are no linux/amd64 nodes available on the kubernetes cluster.")
+                logger.error("This kubernetes cluster doesn't have any nodes with OS 'linux' and architecture 'amd64', for scheduling the Arc-Agents onto and connecting to Azure. Learn more at {}".format("https://aka.ms/ArcK8sSupportedOSArchitecture"))
+        except Exception as ex:
+            tr_logger.error("Couldn't list nodes on the k8s cluster. Error: {}".format(str(ex)))
+
+        try:
+            pod_list = kapi_instance.list_namespaced_pod(consts.Arc_Namespace)
+            pods_count = 0
+            pods_pending = 0
+            for pod in pod_list.items:
+                pods_count += 1
+                tr_logger.debug("Pod {} is in {} state. Reason: {}. Container statuses: {}".format(pod.metadata.name, pod.status.phase, pod.status.reason, pod.status.container_statuses))
+                if pod.status.phase == "Pending":
+                    pods_pending += 1
+            if pods_pending > 0:
+                logger.warning("You have pods stuck in pending state. A possible cause is insuffient resources on the Kubernetes cluster you are trying to onboard to azure-arc. Please run kubectl get pods -n azure-arc to ensure all pods are in running state.")
+            if pods_count == 0:
+                tr_logger.warning("No pods found in azure-arc namespace.")
+
+        except Exception as ex:
+            tr_logger.error("Error occured while fetching pod's statues : {}".format(str(ex)))
+
+        cert_secret = utils.get_kubernetes_secret(kapi_instance, consts.Arc_Namespace, consts.AZURE_IDENTITY_CERTIFICATE_SECRET, custom_logger=tr_logger)
+        if (not cert_secret) or (not hasattr(cert_secret, 'data')) or (consts.AZURE_IDENTITY_CERTIFICATE_SECRET not in cert_secret.data):
+            tr_logger.error("{} secret is not present on the kubernetes cluster.".format(consts.AZURE_IDENTITY_CERTIFICATE_SECRET))
+            logger.error("{} secret is not present on the kubernetes cluster.".format(consts.AZURE_IDENTITY_CERTIFICATE_SECRET))
+
+        try:
+            current_time = datetime.now(timezone.utc)
+            cert_expirn_time = connected_cluster.managed_identity_certificate_expiration_time
+            if cert_expirn_time != datetime.min and cert_expirn_time < current_time:
+                logger.error("MSI certificate on the cluster has expired.")
```

Comment:
CLI output:

"The *whatever name is used in the azure_identity_certificate_secret*on the cluster has expired. This should renew automatically when the cluster has internet connectivity. Please check your internet connectivity." 

Instead of MSI please keep the naming consistent between this and the previous secret not present output. The customer doesn't necessarily need to know about the secret vs. certificate. 

Color: Red

@shashankbarsin can you verify the above.